RELEASE_NOTES.md

```diff
@@ -14,6 +14,29 @@
 
 * The `ExponentialBackoff` and `LinearBackoff` classes now require keyword arguments for their constructor. This change was made to make the classes easier to use and to avoid confusion with the order of the arguments.
 
+* The `BaseApiClient` class is not generic anymore, and doesn't take a function to create the stub. Instead, subclasses should create their own stub right after calling the parent constructor. This enables subclasses to cast the stub to the generated `XxxAsyncStub` class, which have proper `async` type hints. To convert you client:
+
+   ```python
+   # Old
+   class MyApiClient(BaseApiClient[MyServiceStub]):
+       def __init__(self, server_url: str, *, ...) -> None:
+           super().__init__(server_url, MyServiceStub, ...)
+           ...
+
+   # New
```

Comment:
I'd clarify here that `cast` needs to be imported from `typing`

Reply:
I thought about it, but I think it makes the example much more verbose and distracts from the important parts. `cast` is almost a built-in for people using hints, so it should be trivial to figure it out. If you still think it is important, I can add it.

Reply:
Ok, then I think it's fine to keep it as it is

Reply:
I just added it, because I added the import for `MyServiceAsycnStub`, which might be less obvious where should one import it from.